sky/engine/core/script/dart_init.cc

```diff
@@ -544,21 +544,22 @@ void InitDartVM() {
 
   {
     TRACE_EVENT0("flutter", "Dart_Initialize");
-    CHECK(Dart_Initialize(reinterpret_cast<uint8_t*>(
-                              DART_SYMBOL(kDartVmIsolateSnapshotBuffer)),
-                          PrecompiledInstructionsSymbolIfPresent(),
-                          PrecompiledDataSnapshotSymbolIfPresent(),
-                          IsolateCreateCallback,
-                          nullptr,  // Isolate interrupt callback.
-                          nullptr,
-                          IsolateShutdownCallback,
-                          ThreadExitCallback,
-                          // File IO callbacks.
-                          nullptr, nullptr, nullptr, nullptr,
-                          // Entroy source
-                          nullptr,
-                          // VM service assets archive
-                          GetVMServiceAssetsArchiveCallback) == nullptr);
+    char* init_error = Dart_Initialize(
+        reinterpret_cast<uint8_t*>(DART_SYMBOL(kDartVmIsolateSnapshotBuffer)),
+        PrecompiledInstructionsSymbolIfPresent(),
+        PrecompiledDataSnapshotSymbolIfPresent(), IsolateCreateCallback,
+        nullptr,  // Isolate interrupt callback.
+        nullptr, IsolateShutdownCallback, ThreadExitCallback,
+        // File IO callbacks.
+        nullptr, nullptr, nullptr, nullptr,
+        // Entroy source
+        nullptr,
+        // VM service assets archive
+        GetVMServiceAssetsArchiveCallback);
+    LOG_IF(ERROR, init_error != nullptr)
+        << "Error while initializing the Dart VM: " << init_error;
+    CHECK(init_error == nullptr);
+    free(init_error);
```

Comment:
You are correct. But I'd rather keep the `free` in there anyway since `Dart_Initialize` tells us to release the result.

Happy to update this if you have a strong opinion about it.